src/main/java/at/hannibal2/skyhanni/features/garden/contest/JacobContestStatsSummary.kt

```diff
@@ -3,28 +3,129 @@ package at.hannibal2.skyhanni.features.garden.contest
 import at.hannibal2.skyhanni.data.ClickType
 import at.hannibal2.skyhanni.events.CropClickEvent
 import at.hannibal2.skyhanni.events.FarmingContestEvent
+import at.hannibal2.skyhanni.events.GuiRenderEvent
+import at.hannibal2.skyhanni.events.ScoreboardChangeEvent
+import at.hannibal2.skyhanni.events.TabListUpdateEvent
 import at.hannibal2.skyhanni.features.garden.GardenAPI
 import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.LorenzUtils.addAsSingletonList
 import at.hannibal2.skyhanni.utils.LorenzUtils.round
 import at.hannibal2.skyhanni.utils.NumberUtil.addSeparators
+import at.hannibal2.skyhanni.utils.NumberUtil.formatNumber
+import at.hannibal2.skyhanni.utils.NumberUtil.format
+import at.hannibal2.skyhanni.utils.RenderUtils.renderStringsAndItems
+import at.hannibal2.skyhanni.utils.StringUtils.matchMatcher
 import at.hannibal2.skyhanni.utils.TimeUtils
 import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import kotlin.math.roundToInt
+import kotlin.math.roundToLong
 
 class JacobContestStatsSummary {
-    private val config get() = GardenAPI.config
+    private val config get() = GardenAPI.config.jacobContestStats
     private var blocksBroken = 0
     private var startTime = 0L
+    private var startTimeIntoContest = 0.0
+    private var percent = 0.0
+    private var participants = 0
+    private var medalColor = ""
+    private var predictedScore = 0L
+    private var contestStats = emptyList<List<Any>>()
+
+    private val tabContestPattern = " §r(§e○|§6☘) §r§f(?<crop>.+) §r§f◆ §r§f§lTOP §r(?<color>§.)§l(?<percent>(\\d|[.])+)% §r§f◆ (?<participants>.+) ☻".toPattern()
```

Comment:
This should use RepoPattern

Reply:
What's repo pattern?
and should all of the patterns use it?

Reply:
Thats [RepoPattern](https://github.com/hannibal002/SkyHanni/blob/beta/src/main/java/at/hannibal2/skyhanni/utils/repopatterns/RepoPattern.kt)
All three should be updated to the repo
If you have any questions feel free to ask them in contrib chat